oktapam/resource_server_checkout_settings.go

```diff
@@ -0,0 +1,277 @@
+		CheckoutRequired:          plan.CheckoutRequired,
+		CheckoutDurationInSeconds: plan.CheckoutDurationInSeconds.ValueInt32Pointer(),
+		IncludeList:               plan.IncludeList,
+		ExcludeList:               plan.ExcludeList,
+	}
+
+	_, err := r.client.SDKClient.ProjectsAPI.UpdateResourceGroupServerBasedProjectCheckoutSettings(ctx, r.client.Team, plan.ResourceGroup, plan.Project).ResourceCheckoutSettings(*serverCheckoutSettings).Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError("Error updating server checkout settings", err.Error())
+		return
+	}
+
+	// Fetch the updated server checkout settings from the API host
+	updatedServerCheckoutSettings, _, err := r.client.SDKClient.ProjectsAPI.FetchResourceGroupServerBasedProjectCheckoutSettings(ctx, r.client.Team, plan.ResourceGroup, plan.Project).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading server checkout settings",
+			"Could not read server checkout settings for team:"+r.client.Team+"resource_group:"+plan.ResourceGroup+"project_id:"+plan.Project+": "+err.Error(),
+		)
+		return
+	}
+
+	// update the state with the updated server checkout settings
+	plan.CheckoutRequired = updatedServerCheckoutSettings.CheckoutRequired
+	plan.CheckoutDurationInSeconds = types.Int32PointerValue(updatedServerCheckoutSettings.CheckoutDurationInSeconds)
+	plan.IncludeList = updatedServerCheckoutSettings.IncludeList
+	plan.ExcludeList = updatedServerCheckoutSettings.ExcludeList
+
+	// Set state to fully populated data
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete implements resource.Resource.
+func (r *serverCheckoutSettingsResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Retrieve the server checkout settings values from the plan
+	var state serverCheckoutSettingsResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	defaultCheckoutDurationInSeconds := int32(900)
+	// Call the SDK client to reset the server checkout settings to default values
+	serverCheckoutSettings := &pam.ResourceCheckoutSettings{
+		CheckoutRequired:          false,
+		CheckoutDurationInSeconds: &defaultCheckoutDurationInSeconds,
+		IncludeList:               []string{},
+		ExcludeList:               []string{},
+	}
+
+	_, err := r.client.SDKClient.ProjectsAPI.UpdateResourceGroupServerBasedProjectCheckoutSettings(ctx, r.client.Team, state.ResourceGroup, state.Project).ResourceCheckoutSettings(*serverCheckoutSettings).Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError("Error updating server checkout settings", err.Error())
+		return
+	}
+
+	// Set state to empty data
+	state.Id = types.StringValue("")
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// should we delete the state here?
```

Comment:
should we delete the state here?

Reply:
are there docs for the new provider framework that indicate what we should do here?

Reply:
I’ve noticed a pattern in the SDKv2 provider resources where the ID is set to an empty string. There's a recommendation to skip calling the `RemoveResource()` method from the response state . The framework manages this logic automatically if there are no error diagnostics. https://developer.hashicorp.com/terraform/plugin/framework/resources/delete#recommendations Therefore, I believe we’re in good shape here.